### PR TITLE
fix(tup-components, tup-ui): tup-519 UserNews component UI

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

Make "User Updates" more closely match CMS markup and styles.

<sup>Ideally, these lists use the same classes and even markup. Not yet. I'll do that in another PR.</sup>

## Related

- [**TUP-519**](https://jira.tacc.utexas.edu/browse/TUP-519)
- [TUP-532](https://jira.tacc.utexas.edu/browse/TUP-532)
- does **not** fix [TUP-490](https://jira.tacc.utexas.edu/browse/TUP-490) **nor** [TUP-422](https://jira.tacc.utexas.edu/browse/TUP-422)

## Changes

- **added** global `<h3>` styles
    <sup>(probably destined for core-styles)</sup>
- **added** global `[data-prefix]` styles
    <sup>(destined for core-styles)</sup>
- **changed** `UserNews.module.css`
    <sup>(to mimic `.c-feed-list` used by `user_news` app)</sup>
    <sup>(placeholder fixes until portal uses `.c-feed-list`)</sup>
- **changed** `UserNews.tsx`
    <sup>(to mimic markup from `user_news` app)</sup>
    <sup>(to mimic fixes from #249)</sup>
- **added** `maxItems` to store count of User Updates to show
- **changed** "Updated" pill to say "Update"
- **changed** "Posted" pill to say "Published"
- **removed** _inconsistent_ whitespace from title
- **added** _consistent_ whitespace from title

## Testing

1. Compare Portal "User Updates" to "User Updates" plugin (e.g. on [homepage halfway-down](https://tacc.utexas.edu/#user-updates)).
2. Verify only 5 User Updates are shown.
3. Verify date, title, summary, and any "Update" pill in Portal match CMS in:
    - font size
    - spacing
    - html tags
        <sup>(except article title `<h3>` vs. `<h4>` because of page hierarchy)</sup>
	- pill text
	    <sup>(but not pill size yet, because `<Pill>` does not use `c-pill` yet)</sup>
4. Verify space between "Update" pill and article title comes from CSS not whitespace character in title hyperlink.

## UI

| | cms | portal |
| - | - | - |
| date | cms - time | portal -date
| title | cms - title | portal -title
| pill | cms - pill | portal -pill
| summary | cms - summary | portal -summary

## Notes

1. Font in CMS is "Benton Sans". Font in Portal is "Roboto".
2. The "Update" pill in Portal is too small[^1]. Expect fix in [TUP-490](https://jira.tacc.utexas.edu/browse/TUP-490). 

[^1]: Because "Roboto" font is smaller than "Benton Sans" font.

### CODE REVIEW

File: apps/tup-ui/src/main.global.css [hunk 1]
```diff
@@ -45,6 +45,10 @@ h2 {
   font-size: var(--global-font-size--large);
   font-weight: var(--bold);
 }
+h3 {
+  font-size: var(--global-font-size--medium);
+  font-weight: var(--bold);
+}
```
Comment (wesleyboar) [inline apps/tup-ui/src/main.global.css:51]: I am adding these title styles as needed.
<sup>I plan to move them to [Core-Styles > Elements > Headings > Portal](https://tacc.utexas.edu/static/ui/components/detail/headings--portal.html) (currently un-styled) when Core Portal as ready for them.<sup>

File: libs/tup-components/src/news/UserNews.tsx [hunk 2]
```diff
@@ -24,38 +33,44 @@ const ViewAllUpdates = () => (
...
+                {formatDate(newsItem.PostedDate)}
+              </time>
+              <h3 className={styles['title']}>
                 <a
                   href={`/news/user-updates/${newsItem.ID}`}
                   target="_blank"
                   rel="noopener noreferrer"
                 >
-                  {newsItem.WebTitle}
+                  {newsItem.WebTitle.trim()}
```
Comment (wesleyboar) [inline libs/tup-components/src/news/UserNews.tsx:62]: The source data has extra whitespace.

File: libs/tup-components/src/news/UserNews.tsx [hunk 3]
```diff
@@ -24,38 +33,44 @@ const ViewAllUpdates = () => (
...
+
   if (isLoading) return <LoadingSpinner />;
   return (
     <SectionTableWrapper
       header="User Updates"
       headerActions={<ViewAllUpdates />}
       contentShouldScroll
     >
-      <ul className={styles['news-list']}>
+      <div className={styles['news-list']}>
```
Comment (wesleyboar) [inline libs/tup-components/src/news/UserNews.tsx:45]: A list makes sense, but each is also an `<article>`[^1], and `ul` → `li` → `article` is extra nesting that requires undoing browser or future list styles.

[^1]: The `<article>` tag is used for news on CMS in User News plugin and News app. I aim for consistency by doing the same here.

File: libs/tup-components/src/news/UserNews.tsx [hunk 4]
```diff
@@ -24,38 +33,44 @@ const ViewAllUpdates = () => (
 
 const UserNews: React.FC = () => {
   const { data, isLoading } = useUserNews();
+  const maxItems = 5;
```
Comment (wesleyboar) [inline libs/tup-components/src/news/UserNews.tsx:36]: So devs need not fish JSX for a magic number.

**Review by wesleyboar — COMMENTED** [review]
On dev. Tested. Screenshots. ✅

> **Note** The pill alignment and position are fixed in #249.

**Review by jarosenb — APPROVED** [review]
Approved based on dev deployment.

**Pull request merged**